### PR TITLE
Sort assignments on deadline

### PR DESCRIPTION
**Pull request by TimvdLippe**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -214,9 +214,6 @@
             <resource>
                 <directory>src/main/resources</directory>
             </resource>
-            <resource>
-                <directory>target/metamodel</directory>
```
Comment (jwgmeligmeyling) [inline pom.xml:218]: Sure on this one?

File: src/main/java/nl/tudelft/planningstool/api/v1/UserAssignmentAPI.java [hunk 2]
```diff
@@ -34,20 +33,18 @@
...
                 .collect(Collectors.toSet());
 
-        Collection<Assignment> assignments = user.getCourses().stream()
+        List<Assignment> assignments = user.getCourses().stream()
                 .map((c) -> c.getCourse().getAssignments())
                 .flatMap(Collection::stream)
                 .filter(a -> !alreadyPlannedAssignments.contains(a))
-                .filter(a -> a.getDeadline() > now)
-                .collect(Collectors.toSet());
+                .sorted((one, other) -> Long.compare(one.getDeadline(), other.getDeadline()))
```
Comment (jwgmeligmeyling) [inline src/main/java/nl/tudelft/planningstool/api/v1/UserAssignmentAPI.java:46]: I personally like to create a static attribute in `Assignment`:

``` java
public static final Comparator<Assignment> BY_DEADLINE = ((one, other) -> Long.compare(one.getDeadline(), other.getDeadline();
```

That way you can reuse this, and your lambda gets more readable as well: `.sorted(BY_DEADLINE)` :smile:

File: pom.xml (hunk 1 above)
Comment (TimvdLippe) [inline pom.xml:218]: My ide was constantly putting this folder as test resource instead of source. Since th Qclasses are source, or it doesnt compile

Comment (jwgmeligmeyling) [inline pom.xml:218]: Ah very annoying. Yet I hope it didn't break the outputted jar. It needs it's resources :wink: (Not sure if it will break it though!)

Comment (TimvdLippe) [inline pom.xml:218]: Oh thats a good one. #TODO. Can you make an issue for it so I can investigate later? Thanks!

**Pull request merged**